File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& arr) {
        int currsum = arr[0];
        int currsofar = arr[0];
        for(int i=1;i<arr.size();i++)
        {
            if(currsum+arr[i]<arr[i])
            {
                currsum=arr[i];
            }
            else
            {
                currsum+=arr[i];
               
            }
            currsofar = max(currsum,currsofar);
        }
         
        
        return currsofar;
        
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum=nums[0];
        int res=nums[0];
        
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i]+maxSum<nums[i])
            {
                maxSum =nums[i];
            }
            
            else
            {
                maxSum+=nums[i];
            }
            
             res =Math.max(maxSum,res);
        }
        return res;
    }
}